docs/experimental/viz_points_kde.py

```diff
@@ -0,0 +1,277 @@
+
+    Returns
+    -------
+
+    array : np.array
+    Array converted to the given desired range.
+    """
+    if np.max(array) != np.min(array):
+        return ((array - np.min(array))/(np.max(array) - np.min(array)))*(max - min) + min
+    else:
+        raise ValueError(
+            "Can't normalize an array which maximum and minimum value are the same.")
+
+
+kde_dec = """
+float kde(vec3 point, float sigma){
+    return exp(-1.0*pow(length(point), 2.0)/(2.0*sigma*sigma) );
+}
+"""
+
+kde_impl = """
+float current_kde = kde(normalizedVertexMCVSOutput, sigma);
+color = vec3(current_kde);
+fragOutput0 = vec4(color, 1.0);
+"""
+
+tex_dec = """
+vec3 color_mapping(float intensity, sampler2D colormapTexture){;
+    return texture(colormapTexture, vec2(intensity, 0)).rgb;
+}
+"""
+
+tex_impl = """
+vec2 renorm_tex = normalizedVertexMCVSOutput.xy*0.5 + 0.5;
+float intensity = texture(screenTexture, renorm_tex).r;
+
+if(intensity<=0.0){
+    discard;
+}else{
+    color = color_mapping(intensity, colormapTexture).rgb;
+    fragOutput0 = vec4(color, 1.0);
+}
+"""
+
+
+# Windows and scenes setup
+width, height = (1920, 1080)
+offWidth, offHeight = (1080, 1080)
+
+offScene = window.Scene()
+
+off_manager = window.ShowManager(
+    offScene,
+    "demo",
+    (offWidth,
+     offHeight),
+    reset_camera=True,
+    order_transparent=True)
+
+off_manager.window.SetOffScreenRendering(True)
+
+off_manager.initialize()
+
+
+scene = window.Scene()
+scene.set_camera(position=(-6, 5, -10),
+                 focal_point=(0.0,
+                              0.0,
+                              0.0),
+                 view_up=(0.0, 0.0, 0.0))
+
+manager = window.ShowManager(
+    scene,
+    "demo",
+    (width,
+     height),
+    reset_camera=True,
+    order_transparent=True)
+
+
+manager.initialize()
+
+n_points = 1000
+points = np.random.rand(n_points, 3)
+points = normalize(points, -5, 5)
+sigma = 0.25
+scale = 0.5
+
+billboard = actor.billboard(
+    points,
+    (0.0,
+     0.0,
+     1.0),
+    scales=scale,
+    fs_dec=kde_dec,
+    fs_impl=kde_impl)
+
+# Blending and uniforms setup
+shader_apply_effects(off_manager.window, billboard, window.gl_disable_depth)
+shader_apply_effects(off_manager.window, billboard, window.gl_set_additive_blending)
+shader_custom_uniforms(billboard, "fragment").SetUniformf("sigma", sigma)
+
+off_manager.scene.add(billboard)
+
+off_manager.render()
+
+
+# Render to second billboard for color map post-processing.
+textured_billboard = actor.billboard(np.array([[0.0, 0.0, 0.0]]), (1.0, 1.0, 1.0),
+                                     scales=3.0, fs_dec=tex_dec, fs_impl=tex_impl)
+
+cmap = colormaps["viridis"]
+cmap = np.array([cmap(i) for i in np.arange(0.0, 1.0, 1/256)])
+
+
+colormap_to_texture(cmap, "colormapTexture", textured_billboard)
+
+
+window_to_texture(
+    off_manager.window,
+    "screenTexture",
+    textured_billboard,
+    blending_mode="Interpolate")
+
+manager.scene.add(textured_billboard)
+
```

Comment:
If this is a demo or example we would need the usual

```python
if interactive:
   manager.start()

window.record(...)

```

Reply:
This is indeed a demo of this experimental feature I am about to implement as an API, so that is indeed needed, thank you 👍